### PR TITLE
Fix for navbar inputs with Floating Labels.

### PR DESCRIPTION
**Pull request by wallter**
# Before:

b1
b2
b3
b4
# After:

a1
a2
a3
a4

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,7 +15,7 @@ Thumbs.db
 .DS_Store
 /node_modules/
 .grunt/
-/bower_components/
```
Comment (rosskevin) [inline .gitignore:18]: Please revert this file

File: dist/css/bootstrap-material-design.css [hunk 2]
```diff
@@ -2334,24 +2334,54 @@ fieldset[disabled] .navbar .btn-link:focus {
   background-color: #009688;
   color: rgba(255,255,255, 0.84);
 }
-.navbar .navbar-form .form-group input.form-control::-moz-placeholder,
```
Comment (rosskevin) [inline dist/css/bootstrap-material-design.css:2337]: Please revert all dist files, they cloud actual changes.

File: less/_navbar.less [hunk 3]
```diff
@@ -188,13 +188,47 @@
...
+
+      .form-group {
+        &.label-floating,
+        &.label-placeholder {
+          label.control-label {
+            color: @variation-color-text;
+            text-overflow: ellipsis;
+            overflow: hidden;
+            white-space: nowrap;
+            opacity: 0.87;
```
Comment (rosskevin) [inline less/_navbar.less:213]: The majority of this change should not be inside a .generic-variations call.  This creates 4 or so color variations.  If it isn't overriding the color, it should not be inside this block, but above it.

File: less/_navbar.less [hunk 4]
```diff
@@ -188,13 +188,47 @@
...
+            &:not(.is-empty) {
+              label.control-label {
+                top: -23px;
+                line-height: 1.4em;
+                font-size: 10px;
+              }
+            }
+          }
+        }
+      }
```
Comment (rosskevin) [inline less/_navbar.less:235]: any change of the variations needs to mirror the change in the `scss` hierarchy in the relevant scss mixin.  There is usually a comment referencing the specific one.  That's another reason (other than 4 copies) to keep only overrides inside it.

**Comment by wallter** [comment]
@rosskevin - I am not that familiar to less and even less so with `scss`. I would welcome being showed how to rewrite of the few rules I've added - in both less and scss?

I've reverted the `dist/*` files as well as updated the `.gitignore` per your recommendation.

**Comment by rosskevin** [comment]
First, thank you for taking the time to submit a PR.  I encourage you to look further at the less and sass sites, you'll find that they are very much the same, and the nuances are already separated in the code.  Unfortunately, I don't have time to help you further, but you could submit your questions to stackoverflow and reference this pull.  I don't think you are far away from the solution.

**Pull request closed without merging**